src/api/extendOptions.ts

```diff
@@ -29,5 +30,9 @@ function getDefaultStorePath () {
 }
 
 export default (opts?: PnpmOptions): StrictPnpmOptions => {
+  opts = opts || {}
+  if (opts.flatTree === true && !preserveSymlinks) {
+    throw new TypeError('Flat tree is supported only on Node.js >= 6.3.0')
```

Comment:
if expression is little relevant to thrown error, thus it makes it and solution to it a bit obscure. maybe:

```
`--preserve-symlinks` and so 'Flat tree' are not supported on your system, make sure you are running on Node ≽ 6.3.0
```